src/prover.rs

```diff
@@ -747,7 +853,11 @@ fn gpu_prove_from_trace<
     let mut query_details_for_cosets = vec![vec![]; fri_lde_degree];
     let mut query_idx_and_coset_idx_map = vec![0usize; num_queries];
     for query_idx in 0..num_queries {
-        let query_index_lsb_first_bits = bools_buffer.get_bits(&mut transcript, max_needed_bits);
+        let query_index_lsb_first_bits = if is_dry_run()? {
+            vec![(query_idx & 1) == 1; max_needed_bits]
```

Comment:
what's the point of this? just to generate some varying bits?

Reply:
if i would use a constant value, all the queries would target only a single coset, that is undesirable for the dry run and also would blow up on an assertion